Foundation/Host.swift

```diff
@@ -75,7 +75,7 @@ open class Host: NSObject {
     }
     
     open func isEqual(to aHost: Host) -> Bool {
-        return false
+        return aHost._addresses == _addresses && aHost._info == _info && aHost._names == _names && aHost._resolved == _resolved && aHost._type == _type
```

Comment:
`NSHost.isEqual(to:)` is documented as follows:

https://developer.apple.com/documentation/foundation/host/1414786-isequal

> true when the receiver and host share at least one network address; false otherwise.

So this does not seem true. Maybe a naive implementation look like the following?

```swift
if self === other { return true }
return _addresses.firstIndex { aHost._addresses.contains($0) } != nil
```

Reply:
Thank you for watching.
I did not read the document enough.
However, for some reason the test fails or succeeds.
I don't know the cause, I will look a little more closely.

Reply:
The internal `_addresses` is empty before the Host is being resolved, this can be seen in the `open var addresses` which calls `_resolve()` first.

That's why the test fails, @khashiguchi, replacing `_addresses` with `addresses` fixes the problem:
```
open func isEqual(to aHost: Host) -> Bool {
    if self === aHost { return true }
    return addresses.firstIndex { aHost.addresses.contains($0) } != nil
}
```

Reply:
Also the test could be extended to also check for hostnames:
```
func test_isEqual() {
    let host0 = Host(address: "8.8.8.8")
    let host1 = Host(address: "8.8.8.8")
    XCTAssertTrue(host0.isEqual(to: host1))
        
    let hostname0 = Host(name: "google.com")
    let hostname1 = Host(name: "google.com")
    XCTAssertTrue(hostname0.isEqual(to: hostname1))

    let host2 = Host(address: "8.8.8.9")
    XCTAssertFalse(host0.isEqual(to: host2))
    let hostname2 = Host(name: "apple.com")
    XCTAssertFalse(hostname0.isEqual(to: hostname2))
}
```

Reply:
I could not confirm it easily.
Thank you very much, @matis-schotte .
I will fix it early!